server/src/main/java/org/opensearch/index/store/remote/filecache/FileCacheFactory.java

```diff
@@ -37,25 +40,52 @@
  */
 public class FileCacheFactory {
 
-    public static FileCache createConcurrentLRUFileCache(long capacity, CircuitBreaker circuitBreaker) {
-        return new FileCache(createDefaultBuilder().capacity(capacity).build(), circuitBreaker);
+    public static FileCache createConcurrentLRUFileCache(long capacity, CircuitBreaker circuitBreaker, FileTrackerImp fileTrackerImp) {
+        return createFileCache(createDefaultBuilder(fileTrackerImp).capacity(capacity).build(), circuitBreaker);
     }
 
-    public static FileCache createConcurrentLRUFileCache(long capacity, int concurrencyLevel, CircuitBreaker circuitBreaker) {
-        return new FileCache(createDefaultBuilder().capacity(capacity).concurrencyLevel(concurrencyLevel).build(), circuitBreaker);
+    public static FileCache createConcurrentLRUFileCache(
+        long capacity,
+        int concurrencyLevel,
+        CircuitBreaker circuitBreaker,
+        FileTrackerImp fileTrackerImp
+    ) {
+        return createFileCache(
+            createDefaultBuilder(fileTrackerImp).capacity(capacity).concurrencyLevel(concurrencyLevel).build(),
+            circuitBreaker
+        );
     }
 
-    private static SegmentedCache.Builder<Path, CachedIndexInput> createDefaultBuilder() {
+    private static FileCache createFileCache(SegmentedCache<Path, CachedIndexInput> segmentedCache, CircuitBreaker circuitBreaker) {
+        /*
+         * Since OnDemandBlockSnapshotIndexInput.Builder.DEFAULT_BLOCK_SIZE is not overridden then it will be upper bound for max IndexInput
+         * size on disk. A single IndexInput size should always be more than a single segment in segmented cache. A FileCache capacity might
+         * be defined with large capacity (> IndexInput block size) but due to segmentation and concurrency factor, that capacity is
+         * distributed equally across segments.
+         */
+        if (segmentedCache.getPerSegmentCapacity() <= OnDemandBlockSnapshotIndexInput.Builder.DEFAULT_BLOCK_SIZE) {
```

Comment:
This check was gotten rid of recently. I think you need to rebase/remove the check if it was conflicting.